nerfstudio/scripts/render.py

```diff
@@ -129,6 +140,46 @@ def _render_trajectory_video(
                     bounding_box_max = crop_data.center + crop_data.scale / 2.0
                     aabb_box = SceneBox(torch.stack([bounding_box_min, bounding_box_max]).to(pipeline.device))
                 camera_ray_bundle = cameras.generate_rays(camera_indices=camera_idx, aabb_box=aabb_box)
+                if "nearest_camera" in rendered_output_names:
+                    cam_pos = cameras[camera_idx].camera_to_worlds[:, 3].cpu()
+                    cam_rot = R.from_matrix(cameras[camera_idx].camera_to_worlds[:3, :3].cpu())
+                    cam_quat = cam_rot.as_quat()
+
+                    max_dist, max_idx = None, None
+                    true_max_dist, true_max_idx = None, None
+                    for i in range(len(train_cameras)):
+                        train_cam_pos = train_cameras[i].camera_to_worlds[:, 3].cpu()
+                        # Make sure the line of sight from rendered cam to training cam is not blocked by any object
```

Comment:
Is this necessary? I think it would be nice if the result is deterministic, regardless of how the NeRF is. Have you found it to be needed in some cases?

Reply:
I added a flag `check-occlusions` so you can toggle whether or not you want the check